### PR TITLE
[RHPAM-3330] Operator/Image support for Kie-Server Cluster support

### PR DESCRIPTION
**Pull request by desmax74**
See: https://issues.redhat.com/browse/RHPAM-3330
Signed-off-by: desmax74 <mdessi@redhat.com>

Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:

- [x] Pull Request title is properly formatted: `[RHPAM-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Attached commits represent units of work and are properly formatted
- [x] You have read and agreed to the Developer Certificate of Origin (DCO) (see `CONTRIBUTING.md`)
- [x] Every commit contains `Signed-off-by: Your Name <yourname@example.com>` - use `git commit -s`

### CODE REVIEW

File: kieserver/image.yaml [hunk 1]
```diff
@@ -200,6 +200,9 @@ envs:
   - name: "KIE_SERVER_DISABLE_KC_VERIFICATION"
     example: "true"
     description: "Disable Maven KIE Jar verification. It is recommended to test the kjar manually before disabling this verification."
+  - name: "KIE_SERVER_JBPM_CLUSTER"
+    example: "true"
+    description: "Enable the JPM clustering for Kie-Server fail-over, if less than two replica as confiured one additionl replica will be added"
```
Comment (spolti) [inline kieserver/image.yaml:205]:
```suggestion
    description: "Enable the JPM clustering for Kie-Server for processes fail-over. If the number of replicas is less than two, one additional replica will be started."
```

Comment (spolti) [reply kieserver/image.yaml:205]: Hey @desmax74, can you please point me to where this condition is implemented? will it be added on the operator or on the kieserver image?

Comment (desmax74) [reply kieserver/image.yaml:205]: @spolti https://github.com/kiegroup/kie-cloud-operator/pull/589/files#diff-29ad51b906bc49dd54db094a3b626e8e4a04628d03562e6c0d45db0e391ac2fbR705

Comment (spolti) [reply kieserver/image.yaml:205]: ok, thanks.

File: kieserver/image.yaml [hunk 2]
```diff
@@ -202,7 +202,7 @@ envs:
     description: "Disable Maven KIE Jar verification. It is recommended to test the kjar manually before disabling this verification."
   - name: "KIE_SERVER_JBPM_CLUSTER"
     example: "true"
-    description: "Enable the JPM clustering for Kie-Server fail-over, if less than two replica as confiured one additionl replica will be added"
+    description: "Enable the JPM clustering for Kie-Server for processes fail-over. If the number of replicas is less than two, one additional replica will be started."
```
Comment (spolti) [inline kieserver/image.yaml:205]:
```suggestion
    description: "Enable the KIE Server JBPM clustering for processes fail-over. If the number of replicas is less than two, one additional replica will be started."
```

**Review by spolti — DISMISSED** [review]

**Comment by spolti** [comment]
hold until this PR is merged https://github.com/jboss-container-images/jboss-kie-modules/pull/434

File: kieserver/image.yaml [hunk 3]
```diff
@@ -203,6 +203,9 @@ envs:
   - name: "KIE_SERVER_JBPM_CLUSTER"
     example: "true"
     description: "Enable the KIE Server Jbpm clustering for processes fail-over. If the number of replicas is less than two, one additional replica will be started."
+  - name: "KIE_SERVER_JBPM_CLUSTER_TRANSPORT_LOCK_TIMEOUT"
+    value: "60000"
+    description: "Cache configuration on wildfly"
```
Comment (spolti) [inline kieserver/image.yaml:208]:
```suggestion
    description: "Jbpm Cache transport lock timeout, value in milliseconds. The default value is 6000 ms."
```

**Review by spolti — APPROVED** [review]

**Comment by spolti** [comment]
hold until this PR is merged jboss-container-images/jboss-kie-modules#434

File: kieserver/image.yaml [hunk 4]
```diff
@@ -200,6 +200,12 @@ envs:
   - name: "KIE_SERVER_DISABLE_KC_VERIFICATION"
     example: "true"
     description: "Disable Maven KIE Jar verification. It is recommended to test the kjar manually before disabling this verification."
+  - name: "KIE_SERVER_JBPM_CLUSTER"
+    example: "true"
+    description: "Enable the KIE Server Jbpm clustering for processes fail-over. If the number of replicas is less than two, one additional replica will be started."
+  - name: "KIE_SERVER_JBPM_CLUSTER_TRANSPORT_LOCK_TIMEOUT"
+    value: "60000"
+    description: "Jbpm Cache transport lock timeout, value in milliseconds. The default value is 6000 ms."
```
Comment (Kaitou786) [inline kieserver/image.yaml:208]: same comments as in https://github.com/jboss-container-images/rhdm-7-openshift-image/pull/414

Comment (desmax74) [reply kieserver/image.yaml:208]: changed

Comment (spolti) [reply kieserver/image.yaml:208]: the default value on the description seems to be not correct, please fix it and for example, try to use a different value than default.

**Review by spolti — CHANGES_REQUESTED** [review]

**Review by spolti — APPROVED** [review]

**Review by Kaitou786 — APPROVED** [review]

**Pull request merged**